### ISSUE TITLE
Searchable Token suggestion dropdown

### ISSUE THREAD

**Issue by cscheffauer** [tokens-studio/figma-plugin#1437]
**Is your feature request related to a problem? Please describe.**
The problem currently is that if u are mapping tokens to other tokens, the suggestion dropdown doesn't allow searching or typeahead such that a user always needs to either scroll or recall the exact token name.

![image](https://user-images.githubusercontent.com/29776408/202727402-9d4fa77f-880c-4ee4-bebd-63e98ee7a6aa.png)

**Describe the solution you'd like**
A way to search in the suggestion dropdown

**Describe alternatives you've considered**
/

**Additional context**
Not quite sure if this is already possible somehow?

**Comment by six7** [comment]
Thanks for reporting, yeah I think we'll want to do some improvements to that component so you can keep typing. Ideally `suggest` and the dropdown do 2 different things.

Suggest: Autocomplete the current typed value after you typed `{`
Pick: Give a popover that lets me search and pick tokens

**Comment by cscheffauer** [comment]
@six7 thanks - is this something you look into working on?

**Comment by six7** [comment]
> @six7 thanks - is this something you look into working on? 

Yes, we're a bit blocked by #1203 but once that is out of the way there'll be these two modes of suggest/search.

Contributions are always welcome if that's something you'd like to see fixed sooner 🙏

**Comment by cscheffauer** [comment]
> > @six7 thanks - is this something you look into working on?
> 
> Yes, we're a bit blocked by #1203 but once that is out of the way there'll be these two modes of suggest/search.
> 
> Contributions are always welcome if that's something you'd like to see fixed sooner 🙏

Thanks a lot! :) i will have a look at the contribution guide at some point and see if I can help.

**Comment by esreverse** [comment]
@six7  May I ask a question about your first reply
> Pick: Give a popover that lets me search and pick tokens

In my head it would be implemented in a way that all special characters like e.g. "." and "-" will segment a variable, so you could search for: "base.selectable.primary.bg-hover" with a string like "ba pr se ho". Does that sound like your idea of impleenting it?

**Comment by six7** [comment]
I'm closing this in favor of https://github.com/tokens-studio/figma-plugin/issues/1203 which will fulfill those requirements.

@esreverse yes, that'd be ideal! added it to the requirements.
